tests/appstream_glib_validate.pm

```diff
@@ -0,0 +1,39 @@
+# vi: set shiftwidth=4 tabstop=4 expandtab:
+use base 'installedtest';
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+
+    # Test that all the installed AppData files are valid.
+    # If you find that an application is installing an invalid AppData file,
+    # please fix the application *and* suggest that they add their own
+    # build-time `appstream-glib validate` test.
+
+    $self->root_console();
+    assert_script_run('flatpak install --system -y flathub org.freedesktop.appstream-glib');
+    $self->exit_root_console();
+
+    $self->user_console();
+
+    $self->ensure_curl_available();
+
+    # Run it once first to dump the output before potentially failing.
+    script_run('for dir in $(echo $XDG_DATA_DIRS | tr ":" "\n"); do [[ $(compgen -G "${dir}/appdata/*.xml") ]] && flatpak run --file-forwarding org.freedesktop.appstream-glib validate @@ ${dir}/appdata/*.xml >> /var/tmp/appstream-glib-validate.log; done');
```

Comment:
There comes a point where one might want to use some pretty mainstream Perl syntax for multi-line strings:

```suggestion
    script_run(<<'EOF');
for dir in $(echo $XDG_DATA_DIRS | tr ":" "\n"); do
    if [[ $(compgen -G "${dir}/appdata/*.xml") ]]; then
        flatpak run --file-forwarding org.freedesktop.appstream-glib validate @@ ${dir}/appdata/*.xml >> /var/tmp/appstream-glib-validate.log;
    fi;
done
EOF
```